server/resp/src/main/java/org/infinispan/server/resp/RespCommand.java

```diff
@@ -0,0 +1,98 @@
+package org.infinispan.server.resp;
+
+import java.nio.charset.StandardCharsets;
+
+import io.netty.buffer.ByteBuf;
+
+public enum RespCommand {
+   GET,
+   SET,
+   INCR,
+   DECR,
+   DEL,
+   MSET,
+   MGET,
+   PUBLISH,
+   PING,
+   PSUBSCRIBE,
+   PUNSUBSCRIBE,
+   SUBSCRIBE,
+   UNSUBSCRIBE,
+   RESET,
+   COMMAND,
+   ECHO,
+   HELLO,
+   AUTH,
+   CONFIG,
+   INFO,
+   READWRITE,
+   READONLY,
+   SELECT,
+   QUIT;
+
+   private final byte[] bytes;
+
+   RespCommand() {
+      this.bytes = name().getBytes(StandardCharsets.US_ASCII);
+   }
+
+   private static final RespCommand[][] indexedRespCommand;
+
+   static {
+      indexedRespCommand = new RespCommand[26][];
+      // Just manual for now, but we may want to dynamically do this with ordinal determining what order within
+      // a sub array the commands are placed
+      indexedRespCommand[0] = new RespCommand[]{AUTH};
+      indexedRespCommand[2] = new RespCommand[]{CONFIG, COMMAND};
+      indexedRespCommand[3] = new RespCommand[]{DECR, DEL};
+      indexedRespCommand[4] = new RespCommand[]{ECHO};
+      indexedRespCommand[6] = new RespCommand[]{GET};
+      indexedRespCommand[7] = new RespCommand[]{HELLO};
+      indexedRespCommand[8] = new RespCommand[]{INCR, INFO};
+      indexedRespCommand[12] = new RespCommand[]{MGET, MSET};
+      indexedRespCommand[15] = new RespCommand[]{PUBLISH, PING, PSUBSCRIBE, PUNSUBSCRIBE};
+      indexedRespCommand[16] = new RespCommand[]{QUIT};
+      indexedRespCommand[17] = new RespCommand[]{RESET, READWRITE, READONLY};
+      indexedRespCommand[18] = new RespCommand[]{SET, SUBSCRIBE, SELECT};
+      indexedRespCommand[20] = new RespCommand[]{UNSUBSCRIBE};
+   }
+
+   public static RespCommand fromByteBuf(ByteBuf buf, int commandLength) {
+      if (buf.readableBytes() < commandLength + 2) {
+         return null;
+      }
+      int readOffset = buf.readerIndex();
```

Comment:
Is this check necessary? Intrinsic is asserting the size + separator.

Reply:
Which check sorry? You mean the if above? If so then technically it is not as all the callers have verified it is there, but I was keeping this separated from Intrinsic. Let me think on if we want to document is requiring the bytes or not, either way this check is one of the cheapest things we can do.